src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -228,17 +219,17 @@ private IServiceCollection BuildHostingServices()
             return services;
         }
 
-        private string ResolveApplicationBasePath(string applicationBasePath, string basePath)
+        private string ResolveContentRootPath(string contentRootPath, string basePath)
         {
-            if (string.IsNullOrEmpty(applicationBasePath))
+            if (string.IsNullOrEmpty(contentRootPath))
```

Comment:
The app base path defaults to where the executable lives and is executed. If you set the web.config processPath to point to the executable, there is no problem with app base path or any need of an environment variable ... ie ... I think @moozzyk should reconsider how publish-IIS is setup, but I'm fairly certain he has, so maybe I just need to trust him on this.